File: C#/guess.cs
using System;

namespace MyApp;

class Program2 {
    public static void guess() {
        
        Random random = new Random();
        bool playAgain = true;
        int min = 1;
        int max = 100;
        int guess;
        int number;
        int guesses;

        while (playAgain) {
            Console.WriteLine("------------------------------------");
            Console.WriteLine("Welcome to the Number Guessing Game!");
            Console.WriteLine("------------------------------------");
            
            number = random.Next(min, max + 1);
            guess = 0;
            guesses = 0;

            Console.WriteLine($"Guess a number between {min} and {max}:");

            while (guess != number) {
                string input = Console.ReadLine();
                if (int.TryParse(input, out guess)) {
                    guesses++;
                    if (guess < number) {
                        Console.WriteLine("Too low! Try again:");
                    } else if (guess > number) {
                        Console.WriteLine("Too high! Try again:");
                    } else {
                        Console.WriteLine($"Congratulations! You've guessed the number {number} in {guesses} attempts.");
                    }
                } else {
                    Console.WriteLine("Please enter a valid number:");
                }
            }

            Console.WriteLine("Do you want to play again? (yes/no)");
            string response = Console.ReadLine().ToLower();
            playAgain = response == "yes";
        }

    }
}

File: C++/Tutorial/guess.cpp
#include <iostream>
#include <ctime>


int main() {

    int guess;
    int tries = 0;
    int num;

    srand(time(NULL)); 
    num = (rand() % 100) + 1;

    std::cout << "Welcome to the Guessing Game!" << '\n';

    do {
        std::cout << "Enter your guess (1-100): ";
        std::cin >> guess;
        tries++;
        if (guess > num) {
            std::cout << "Too high! Try again." << '\n';
        } else if (guess < num) {
            std::cout << "Too low! Try again." << '\n';
        } else if (guess == num) {
            std::cout << "Congratulations! You guessed the number in " << tries << " tries." << '\n';
        } else {
            std::cout << "Invalid input! Please enter a number between 1 and 100." << '\n';
            tries--;
        }

    } while (guess != num);

    return 0;
}